sgx-show/src/main.rs

```diff
@@ -0,0 +1,176 @@
+//! This program exists to gather and display data relating to SGX hardware,
+//! SGX BIOS enablement, SGX kernel drivers and potentially SGX-related
+//! software for use in debugging.
+//!
+//! It works by running a hierarchical tree of tests and displaying the
+//! information gathered therefrom. If a parent test fails to validate,
+//! the child test are still called to produce their output, but do not
+//! perform data collection since the data collection method may be invalid.
+//! For example, we don't attempt to look up CPUIDs for SGX attributes if
+//! the CPU doesn't support SGX.
+//!
+//! Each test has a name, as well as a `data` source, a data `sink` and
+//! a (possibly empty) array of child tests. First, it attempts to collect
+//! the data. Second, it converts the input data to the data type for the
+//! sink. Third, it passes the data to the sink to determine if this data
+//! is successful or not. Fourth, it prints the succes status, the test
+//! name, the data source and the sink to the command line. Finally, it
+//! calls all child tests, passing the status of the parent test to the
+//! child.
+//!
+//! Note that there are differet kinds of sinks. Some perform data validation.
+//! Others, like `sink::debug::Debug` just dump some of the input data.
+//!
+//! There is currently only one data source (CPUID).
+
+mod data;
+mod exec;
+mod sink;
+
+use exec::*;
+use sgx_types::{attr, misc};
+
+use data::cpuid::{CpuId, CpuInfo, ExtFeat, SgxAttr, SgxCaps};
+use sink::{bit::Bit, debug::Debug, mask::Mask};
+
+pub trait Convert<T>: Sized {
```

Comment:
Could we use the existing https://doc.rust-lang.org/std/convert/trait.TryInto.html instead here?

Reply:
Originally, no; which is why I created `Convert`. But because of other changes that is now possible. I have moved this to `TryFrom`. So: fixed.